test/test_torch.py

```diff
@@ -15147,6 +15147,17 @@ def test_full_out(self, device):
         # verifies out dtype overrides inference
         self.assertEqual(torch.full(o.shape, 1., out=o).dtype, o.dtype)
 
+    # Checks that float->integer casts don't produce undefined behavior errors
+    # Note: in CPP, casting from a floating point value to an integral dtype
+    # is undefined if the floating point value is not within the integral
+    # dtype's range. This can (and should) cause undefined behavior errors
+    # with ASAN/UBSAN. This behavior is expected in PyTorch, however,
+    # and is acceptable even if undefined.
+    @dtypes(torch.bool, torch.uint8, torch.int8, torch.int16, torch.int32, torch.int64)
+    def test_float_to_int_undefined_conversion(self, device, dtype):
+        t = torch.tensor((-9.2234e+18, 9.2234e+18), dtype=torch.float)
```

Comment:
device=device

Reply:
how'd you come up with this value?  This looks like MAX_INT64 but I'm not sure what this turns out to be represented as in float.  Isn't there some std::limits thing you can use that you know can be represented?

Reply:
Great question.

The max int64_t value is 2**63-1 = 9,223,372,036,854,775,807. PyTorch will actually throw an error if you try to create a float tensor with this value (yikes!), but NumPy will happily make one:

```
np.array((2**63,), dtype=np.float)
: [9.22337204e+18]
```

We can then put that value into PyTorch:

```
torch.tensor((9.22337204e+18,), dtype=torch.float)
: tensor([9.2234e+18])
```

And creating a float tensor with +/- 9.2234e+18 finally gives us identity in PyTorch.

Reply:
> Isn't there some std::limits thing you can use that you know can be represented?

Not that I'm familiar with in Python.

Reply:
Greg suggesting just using the equivalent of std::numeric_limits<float>::max, which is 3.402823466e+38. I'll make the change.